### PR TITLE
MOTECH-2760: Added attributes for create/update Person task actions

### PR DESCRIPTION
**Pull request by mkruszynski**

### CODE REVIEW

**Comment by sebbrudzinski** [comment]
test this please

**Comment by sebbrudzinski** [comment]
test this please

File: openmrs/src/main/java/org/motechproject/openmrs/resource/PersonResource.java [hunk 1]
```diff
@@ -50,6 +50,16 @@
     AttributeTypeListResult queryPersonAttributeTypeByName(Config config, String name);
 
     /**
+     * Returns {@code Attribute.AttributeType} of the attribute matching given uuid. The given {@code config} will
+     * be used while performing this action.
+     *
+     * @param config  the configuration to be used while performing this action
+     * @param uuid the uuid of the AttributeType
```
Comment (ngraczewski) [inline openmrs/src/main/java/org/motechproject/openmrs/resource/PersonResource.java]: User two spaces to separate the variable name and description.

File: openmrs/src/main/java/org/motechproject/openmrs/resource/PersonResource.java [hunk 2]
```diff
@@ -50,6 +50,16 @@
...
+     * @return
```
Comment (ngraczewski) [inline openmrs/src/main/java/org/motechproject/openmrs/resource/PersonResource.java]: Missing description.

File: openmrs/src/main/java/org/motechproject/openmrs/resource/PersonResource.java [hunk 3]
```diff
@@ -89,11 +99,32 @@
...
     void deletePerson(Config config, String personUuid);
+
+    /**
+     * Checking Person's AttributeType format. If format given by OpenMRS server is other than standard Java types
+     * (format is internal openMRS object), that means attributes JSON must have hydratedObject field, not value.
+     * Person Attribute type can be java.lang.String/Boolean/Integer/Float or Character. Internal OpenMRS formats
+     * are e.g. org.openmrs.Concept or org.openmrs.Person
+     *
+     * @param config  the configuration to be used while performing this action
+     * @param person Person that will be created in OpenMRS.
```
Comment (ngraczewski) [inline openmrs/src/main/java/org/motechproject/openmrs/resource/PersonResource.java]: User two spaces to separate the variable name and description. Also, description should start with lower case.

File: openmrs/src/main/java/org/motechproject/openmrs/tasks/OpenMRSActionProxyService.java [hunk 4]
```diff
@@ -63,15 +63,16 @@ void createEncounter(String configName, DateTime encounterDatetime, String encou
      * @param motechId  the patient motechId
      * @param locationForMotechId  the location name for identifiers
      * @param identifiers  the additional identifiers to be stored to patient
+     * @param personAttributes the additional attributes of person
```
Comment (ngraczewski) [inline openmrs/src/main/java/org/motechproject/openmrs/tasks/OpenMRSActionProxyService.java]: User two spaces to separate the variable name and description.

File: openmrs/src/main/java/org/motechproject/openmrs/tasks/OpenMRSActionProxyService.java [hunk 5]
```diff
@@ -112,14 +113,15 @@ Patient createPatient(String configName, String givenName, String middleName, St
      * @param gender  the person gender
      * @param dead  the person dead flag
      * @param causeOfDeathUUID  the cause of the death
+     * @param personAttributes the additional attributes of person
```
Comment (ngraczewski) [inline openmrs/src/main/java/org/motechproject/openmrs/tasks/OpenMRSActionProxyService.java]: User two spaces to separate the variable name and description.

File: openmrs/src/main/java/org/motechproject/openmrs/tasks/OpenMRSActionProxyService.java [hunk 6]
```diff
@@ -34,7 +34,7 @@ void createEncounter(String configName, DateTime encounterDatetime, String encou
     /**
      * Creates a patient with the given params. The required fields are : {@code givenName}, {@code familyName},
      * {@code gender}, {@code motechId}. If the locationName is not provided, the default location will be used.
-     * Configuration with the given {@code configName} will be used while performing this action. Returns created Paitent.
+     * Configuration with the given {@code configName} will be used while performing this action. Returns created Patient.
```
Comment (ngraczewski) [inline openmrs/src/main/java/org/motechproject/openmrs/tasks/OpenMRSActionProxyService.java]: patient

**Comment by mkruszynski** [comment]
test this please

**Pull request merged**